File: 20190917/20190920/118. 杨辉三角.cc
/*
 * Description: 杨辉三角
 * Author: AskeyNil
 * CreateDate: 2019-09-20 08:04:44
 * LastEditors: AskeyNil
 *
 * *********************************
 * **                             **
 * **     　  你只有足够努力     　  **
 * **       才能看上去毫不费力       **
 * **                             **
 * *********************************
 *
 */

#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Solution {
   public:
    vector<vector<int>> generate(int numRows) {
        if (numRows == 0) {
            vector<vector<int>> vv;
            return vv;
        } else {
            vector<vector<int>> vv = this->generate(numRows - 1);
            vv.push_back(vector<int>());
            if (numRows != 1) {
                int left = 0;
                for (auto num : vv[numRows - 2]) {
                    vv[numRows - 1].push_back(left + num);
                    left = num;
                }
            }
            vv[numRows - 1].push_back(1);
            return vv;
        }
    }
};

File: 20190917/20190920/118. 杨辉三角.py
'''
Description: 杨辉三角
Author: AskeyNil
CreateDate: 2019-09-20 07:45:07
LastEditors: AskeyNil

*********************************
**                             **
**     　  你只有足够努力     　  **
**       才能看上去毫不费力       **
**                             **
*********************************

'''


class Solution:
    def generate(self, numRows: int) -> [[int]]:
        if numRows == 0:
            return [[]
        if numRows == 1:
            return [[1]]
        else:
            left = 0
            rows = self.generate(numRows - 1)
            rows.append([])
            for num in rows[-2]:
                rows[-1].append(num + left)
                left = num
            rows[-1].append(1)
            return rows


print(Solution().generate(100))
